Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/utils/VirtualReality.java

```diff
@@ -0,0 +1,150 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+package org.dolphinemu.dolphinemu.utils;
+
+import android.content.Context;
+import android.content.Intent;
+import android.content.pm.ApplicationInfo;
+import android.content.pm.PackageManager;
+import android.net.Uri;
+import android.os.Build;
+import android.os.Bundle;
+
+import org.dolphinemu.dolphinemu.BuildConfig;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.ArrayList;
+import java.util.Locale;
+
+public class VirtualReality
+{
+  private static final String VR_PACKAGE = "org.dolphinemu.dolphinemu.vr";
+
+  public static boolean isActive()
+  {
+    return BuildConfig.BUILD_TYPE.equals("vr");
+  }
+
+  public static boolean isInstalled(Context context)
+  {
+    PackageManager pm = context.getPackageManager();
+    for (ApplicationInfo app : pm.getInstalledApplications(PackageManager.GET_META_DATA))
+    {
+      if (app.packageName.equals(VR_PACKAGE))
+      {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  public static boolean isLegacyPath(String filePath)
+  {
+    return filePath.startsWith("/");
+  }
+
+  public static void linkLoader()
+  {
+    if (isActive())
+    {
+      String manufacturer = Build.MANUFACTURER.toLowerCase(Locale.ROOT);
+      // rename oculus to meta as this will probably happen in the future anyway
+      if (manufacturer.contains("oculus"))
+        manufacturer = "meta";
+
+      //Load manufacturer specific loader
+      try
+      {
+        System.loadLibrary("openxr_" + manufacturer);
+      }
+      catch (UnsatisfiedLinkError e)
+      {
+        Log.error("Unsupported VR device: " + manufacturer);
+        System.exit(0);
+      }
+    }
+  }
+
+  public static void openIntent(Context context, String[] filePaths)
+  {
+    Intent launcher = context.getPackageManager().getLaunchIntentForPackage(VR_PACKAGE);
+
+    if (VirtualReality.isLegacyPath(filePaths[0]))
+    {
+      launcher.putExtra("AutoStartFiles", filePaths);
+    }
+    else
+    {
+      Uri uri;
+      try
+      {
+        uri = ContentHandler.unmangle(filePaths[0]);
+      }
+      catch (FileNotFoundException e)
+      {
+        uri = Uri.parse(filePaths[0]);
+      }
+      launcher.addFlags(Intent.FLAG_GRANT_PERSISTABLE_URI_PERMISSION);
```

Comment:
There shouldn't be any need for persistable. It's only going to run the game as a response to the intent, not permanently save the URI to the game somewhere on disk, right?